src/go/privileged-service/pkg/port/portproxy_test.go

```diff
@@ -0,0 +1,316 @@
+		PortMap: nat.PortMap{
+			"80/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "80",
+				},
+			},
+		},
+		ConnectAddrs: []types.ConnectAddrs{
+			{Network: "tcp", Addr: "192.168.0.1"},
+			{Network: "tcp", Addr: "192.168.0.2"},
+			{Network: "tcp", Addr: "192.168.0.3"},
+		},
+	}
+
+	ports6 := portProxy{
+		PortMap: nat.PortMap{
+			"80/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "80",
+				},
+			},
+		},
+		ConnectAddrs: []types.ConnectAddrs{
+			{Network: "tcp", Addr: "192.168.0.2"},
+			{Network: "tcp", Addr: "192.168.0.1"},
+			{Network: "tcp", Addr: "192.168.0.3"},
+		},
+	}
+
+	ports7 := portProxy{
+		PortMap: nat.PortMap{
+			"443/tcp": []nat.PortBinding{
+				{
+					HostIP:   "192.168.0.10",
+					HostPort: "443",
+				},
+			},
+		},
+		ConnectAddrs: wslConnectAddrs,
+	}
+
+	ports8 := portProxy{
+		PortMap: nat.PortMap{
+			"443/tcp": []nat.PortBinding{
+				{
+					HostIP:   "192.168.0.11",
+					HostPort: "443",
+				},
+			},
+		},
+		ConnectAddrs: wslConnectAddrs,
+	}
+	ports9 := portProxy{
+		PortMap: nat.PortMap{
+			"80/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "80",
+				},
+				{
+					HostIP:   "127.0.0.2",
+					HostPort: "80",
+				},
+				{
+					HostIP:   "127.0.0.3",
+					HostPort: "80",
+				},
+			},
+			"443/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "443",
+				},
+				{
+					HostIP:   "127.0.0.3",
+					HostPort: "443",
+				},
+				{
+					HostIP:   "127.0.0.2",
+					HostPort: "443",
+				},
+			},
+		},
+		ConnectAddrs: []types.ConnectAddrs{
+			{Network: "tcp", Addr: "192.168.0.2"},
+			{Network: "tcp", Addr: "192.168.0.1"},
+			{Network: "tcp", Addr: "192.168.0.3"},
+		},
+	}
+	ports10 := portProxy{
+		PortMap: nat.PortMap{
+			"443/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.2",
+					HostPort: "443",
+				},
+				{
+					HostIP:   "127.0.0.3",
+					HostPort: "443",
+				},
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "443",
+				},
+			},
+			"80/tcp": []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.2",
+					HostPort: "80",
+				},
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: "80",
+				},
+				{
+					HostIP:   "127.0.0.3",
+					HostPort: "80",
+				},
+			},
+		},
+		ConnectAddrs: []types.ConnectAddrs{
+			{Network: "tcp", Addr: "192.168.0.1"},
+			{Network: "tcp", Addr: "192.168.0.2"},
+			{Network: "tcp", Addr: "192.168.0.3"},
+		},
+	}
+
+	ports11 := portProxy{
+		PortMap:      nil,
+		ConnectAddrs: nil,
+	}
+
+	ports12 := portProxy{
+		PortMap:      nil,
+		ConnectAddrs: nil,
+	}
+
+	ports13 := portProxy{
+		PortMap:      make(nat.PortMap),
+		ConnectAddrs: []types.ConnectAddrs{},
+	}
+
+	ports14 := portProxy{
+		PortMap:      make(nat.PortMap),
+		ConnectAddrs: []types.ConnectAddrs{},
+	}
+
+	tests := []struct {
+		description string
+		actual      portProxy
+		expect      portProxy
+		shouldMatch bool
+		expectedErr error
```

Comment:
This can't be anything non-`nil`, because we expect _both_ hashes to lead to the same error. (This means the coverage is not great.)